repository/src/test/java/org/alfresco/schedule/AbstractScheduledLockedJobTest.java

```diff
@@ -0,0 +1,208 @@
+/*
+ * #%L
+ * Alfresco Repository
+ * %%
+ * Copyright (C) 2005 - 2023 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software. 
+ * If the software was purchased under a paid Alfresco license, the terms of 
+ * the paid license agreement will prevail.  Otherwise, the software is 
+ * provided under the following open source license terms:
+ * 
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * 
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.schedule;
+
+import org.alfresco.model.ContentModel;
+import org.alfresco.repo.model.Repository;
+import org.alfresco.repo.security.authentication.AuthenticationUtil;
+import org.alfresco.repo.transaction.RetryingTransactionHelper;
+import org.alfresco.service.cmr.repository.ChildAssociationRef;
+import org.alfresco.service.cmr.repository.NodeRef;
+import org.alfresco.service.cmr.repository.NodeService;
+import org.alfresco.service.cmr.repository.StoreRef;
+import org.alfresco.service.namespace.NamespaceService;
+import org.alfresco.service.namespace.QName;
+import org.alfresco.service.namespace.RegexQNamePattern;
+import org.alfresco.service.transaction.TransactionService;
+import org.alfresco.util.BaseSpringTest;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.junit.Before;
+import org.junit.Test;
+import org.quartz.JobDetail;
+import org.quartz.SchedulerException;
+import org.springframework.scheduling.quartz.SchedulerAccessorBean;
+import org.springframework.test.context.ContextConfiguration;
+
+import com.google.common.collect.ImmutableMap;
+
+/**
+ * 
+ * @author Tiago Salvado
+ */
+@ContextConfiguration({"classpath:alfresco/application-context.xml", "classpath:alfresco/schedule/test-schedule-context.xml"})
+public class AbstractScheduledLockedJobTest extends BaseSpringTest
+{
+    private static final int TOTAL_NODES = 9;
+    private static final int NUM_THREADS = 2;
+    private static final long JOB_EXECUTER_LOCK_TTL = 30000L;
```

Comment:
Does this need to be as long as 30 seconds?  Can we make it e.g. 1 second for the test?

Reply:
This needs to be the same time as [ScheduledJobLockExecuter.LOCK_TTL](https://github.com/Alfresco/alfresco-community-repo/blob/ffe60440af4971c89f3a21e8ce99ff1777123f20/repository/src/main/java/org/alfresco/schedule/ScheduledJobLockExecuter.java#L57) otherwise the test won't cover the original problem.